xlators/mgmt/glusterd/src/snapshot/glusterd-lvm-snapshot.c

```diff
@@ -955,9 +955,20 @@ glusterd_lvm_snap_clone_brick_path(char *snap_mount_dir,
 
     if ((len < 0) || (len >= sizeof(brick_path))) {
         ret = -1;
+        goto out;
     }
 
-    *snap_brick_path = brick_path;
+    *snap_brick_path = gf_strdup(brick_path);
+    if (!*snap_brick_path) {
+        ret = -1;
+        goto out;
```

Comment:
We can avoid the goto statement here.

Reply:
In success case it will become a leak, we need to correct it.

Reply:
@mohit84 I think the freeing part is supposed to be handled by the callee? As if we free it on success the callee might get a NULL pointer deref, no?

Reply:
> We can avoid the goto statement here.

Couple of reasons why I added the `goto`:
- If there are some lines of code to be added in future after the `if` statement.
- Code readability. You see `ret` being set to `-1` and `goto` keyword and you know some error has occurred.
- Doesn't really impact performance as the `goto` block comes right after the end of `if` block.

Anyways, should I go ahead and remove it?

Reply:
> @mohit84 I think the freeing part is supposed to be handled by the callee? As if we free it on success the callee might get a NULL pointer deref, no?

Can you confirm if the caller functions are handling the freeing of the pointer or not?
If so, we can decide on it accordingly.

Reply:
> > We can avoid the goto statement here.
> 
> Couple of reasons why I added the `goto`:
> 
> * If there are some lines of code to be added in future after the `if` statement.
> * Code readability. You see `ret` being set to `-1` and `goto` keyword and you know some error has occurred.
> * Doesn't really impact performance as the `goto` block comes right after the end of `if` block.

This was the main reason for avoiding it.

> Anyways, should I go ahead and remove it?

Anyways, it doesn't create much noise which could be a concern. You can let it be as it is.

Reply:
I think we don't need to allocate memory to populate path. We can directly pass the brickinfo as an argument by the parent function and copy the  brickpath to  brickinfo->path Instead of passing snap_brick_path.

Reply:
CC @aravindavk

Reply:
@mohit84 can we keep the current fix as is? (ie, resolve coverity CWE first) and take up that optimization as a followup? Reason, I would prefer to get zfs based snapshot tests running before getting to make changes in snapshot code. Lets see.

Reply:
> @mohit84 can we keep the current fix as is? (ie, resolve coverity CWE first) and take up that optimization as a followup? Reason, I would prefer to get zfs based snapshot tests running before getting to make changes in snapshot code. Lets see.

It will introduce a new coverity bug.

Reply:
Posted a new patch https://github.com/gluster/glusterfs/pull/3094 to resolve the same.